src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectGrantController.java

```diff
@@ -266,15 +346,18 @@ private CatalogObjectGrantMetadata updateCatalogObjectGrantForAUser(
                             @ApiResponse(code = 403, message = "Permission denied"), })
     @RequestMapping(value = REQUEST_API_QUERY + "/group", method = PUT)
     @ResponseStatus(HttpStatus.OK)
-    private CatalogObjectGrantMetadata updateCatalogObjectGrantForAGroup(
+    @Transactional
+    public CatalogObjectGrantMetadata updateCatalogObjectGrantForAGroup(
             @ApiParam(value = "The session id used to access ProActive REST server.", required = true) @RequestHeader(value = "sessionID", required = true) String sessionId,
             @ApiParam(value = "The name of the bucket where the catalog object is stored.", required = true) @PathVariable String bucketName,
             @ApiParam(value = "The name of the object in the bucket, which is the subject of the grant.", required = true) @PathVariable String catalogObjectName,
-            @ApiParam(value = "The new type of the access grant. It can be either read, write or admin.", required = true) @RequestParam(value = "accessType", required = true) String accessType,
-            @ApiParam(value = "The name of the group of users that are benefiting of the access grant.", defaultValue = "") @RequestParam(value = "userGroup", required = true, defaultValue = "") String userGroup)
+            @ApiParam(value = "The new type of the access grant. It can be either noAccess, read, write or admin.", required = true) @RequestParam(value = "accessType", required = true) String accessType,
+            @ApiParam(value = "The new priority of the access grant. It can be a value from 1 to 10.", required = true) @RequestParam(value = "priority", required = true) int priority,
```

Comment:
The new priority of the access grant. It can be a value from 1 (lowest) to 10 (highest), with 5 as default.
Priorities are used to compute the final acces rights of a user belonging to multiple groups. Group grants with the same priority will resolve with the default accessType order (admin > write > read > noaccess).
Finally, The general decision mechanism to compute an object access rights is userObjectGrant > userBucketGrant > groupsObjectGrant > groupsBucketGrant.